spec/models/tour_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'rails_helper'
+
+RSpec.describe Tour, type: :model do
+  let(:category) {
+    Category.create!(
+      name: "Example category"
+    )
+  }
+
+  let(:tour_1){
+    Tour.create!(
+      name: "Name Tour1 ",
+      description: "abc",
+      start_date: Time.zone.parse('2022-09-11 21:00'),
+      end_date: Time.zone.parse('2022-10-11 21:00'),
+      category_id: category.id,
+      price: 1,
+      stock: 1,
+      avg_rating: 1
+    )
+  }
+
+  let(:tour_2){
+    Tour.create!(
+      name: "Name Tour2 ",
+      description: "abc",
+      start_date: Time.zone.parse('2022-10-11 21:00'),
+      end_date: Time.zone.parse('2022-11-11 21:00'),
+      category_id: category.id,
+      price: 2,
+      stock: 1,
+      avg_rating: 1
+    )
+  }
+
+  describe "presence" do
+    it { should validate_presence_of(:name)}
+    it { should validate_presence_of(:description)}
+    it { should validate_presence_of(:start_date)}
+    it { should validate_presence_of(:price)}
+    it { should validate_presence_of(:stock)}
+    it { should validate_presence_of(:avg_rating)}
+  end
+
+  describe "check scope" do
+    it "orders by id ascending" do
```

Comment:
describe, it, context là gì, dùng như thế nào ?

Reply:
describe nhóm các test case theo model, controller, method, class
context dùng để nhóm các điều kiện của example
it định nghĩa 1 test case